scripts/multisig.js

```diff
@@ -0,0 +1,140 @@
+        return accountId.endsWith('.testnet');
+    }
+
+    return accountId.endsWith('.near');
+}
+
+function isEmail(detail) {
+    return detail.includes('@');
+}
+
+function isEquivalentPhoneNumber(phone1, phone2) {
+    const nonDigits = /\D/g;
+    return phone1.replace(nonDigits, '') === phone2.replace(nonDigits, '');
+}
+
+function printRecoveryMethods(recoveryMethods) {
+    recoveryMethods
+        .sort(({ kind: kindA }, { kind: kindB }) => {
+            if (kindA > kindB) {
+                return 1;
+            }
+            return -1;
+        })
+        .forEach(({ detail, kind }) => {
+            const is2fa = is2faRecoveryMethod(kind);
+            if (kind === RECOVERY_METHOD_KINDS.PHRASE || kind === RECOVERY_METHOD_KINDS.LEDGER) {
+                console.log(`[${kind}]`);
+            } else {
+                console.log(`${is2fa ? '*' : ''}[${kind}]: ${detail}`);
+            }
+        });
+}
+
+async function lookupRecoveryMethods(accountId) {
+    if (!isAccountValid(accountId)) {
+        console.error(`Invalid account ID ${accountId}`);
+        return;
+    }
+
+    const recoveryMethods = await listRecoveryMethodsByAccountId(accountId)
+        .map(({ detail, kind }) => ({ detail, kind }));
+
+    if (!recoveryMethods.length) {
+        console.log(`\n\nNo recovery methods found for ${accountId}`);
+        return;
+    }
+
+    console.log(`\n\nFound ${recoveryMethods.length} recovery method(s) for ${accountId}:`);
+    printRecoveryMethods(recoveryMethods);
+}
+
+async function transferMultisig({ accountId, current, desired }) {
+    if (!isAccountValid(accountId)) {
+        console.error(`Invalid account ID ${accountId}`);
+        return;
+    }
+
+    if (!isEmail(desired)) {
+        console.error(`Invalid email: ${desired}. The 2FA method can only be set to email.`);
+        return;
+    }
+
+    const isCurrentPhone = !isEmail(current);
+    const currentMethodKind = isCurrentPhone ? TWO_FACTOR_AUTH_KINDS.PHONE : TWO_FACTOR_AUTH_KINDS.EMAIL;
+    const [sms2faMethod] = await listRecoveryMethodsByAccountId(accountId)
+        .filter(({ detail, kind }) => kind === currentMethodKind && (isCurrentPhone ? isEquivalentPhoneNumber(detail, current) : (detail === current)));
+
+    if (!sms2faMethod) {
+        console.error(`No 2FA recovery method found for account ${accountId} with [${currentMethodKind}] ${current}`);
+        return;
+    }
+
+    const { confirmUpdate } = await inquirer.prompt({
+        name: 'confirmUpdate',
+        type: 'confirm',
+        message: `
+            This will update ${accountId}'s current 2FA recovery method for ${current} to an email 2FA recovery method for ${desired}. Proceed?
+        `,
+    });
+
+    if (!confirmUpdate) {
+        console.warn('Aborting');
+        return;
+    }
+
+    await deleteRecoveryMethod({
+        accountId,
+        kind: currentMethodKind,
+    });
+
+    const email2faMethod = await updateRecoveryMethod({
```

Comment:
At this point we'd actually be creating a new document rather that updating right? I know `updateRecoveryMethod` will create if it doesn't exist, curious if there is any reason to use this over `createRecoveryMethod`?

Reply:
Also do we need to be concerned about `deleteRecoveryMethod` succeeding and this (creation) failing? should we `catch` errors thrown from this and add some specific logging to let CSA know something has gone wrong and will require action from an engineer? Maybe printing the original document would be useful here?

Reply:
> At this point we'd actually be creating a new document rather that updating right?
Good point, I had the update in there before I realized that the library wouldn't delete the current document as part of updating the range key.

However since I made the change to support email-email transfers I think I will leave the update but make the delete contingent on whether the range key is updated.

> Also do we need to be concerned about `deleteRecoveryMethod` succeeding and this (creation) failing?
Not if it remains as an update I think. Good call on printing the original document upfront though, I'll add that.